### DISCUSSION TITLE
Displaying extended fields on the admin-api

### DISCUSSION THREAD

**Discussion by guillmelo** [medusajs/medusa#3918] [Q&A]
Hello, I followed the steps on https://docs.medusajs.com/development/entities/extend-entity, I can see the fields on my database, however I cannot see them on the admin-api, am I missing something?

Thanks

**Comment by olivermrbl** [comment]
@guillmelo – can I get you to ensure, you've [added the custom fields to the set of default response properties](https://docs.medusajs.com/development/entities/extend-entity#access-custom-attributes-and-relations-in-core-endpoints)?

**Reply by guillmelo** [reply]
Hello @olivermrbl, thanks for the response, yes I added the response properties and the allowedStoreProductsRelations and
defaultStoreProductsRelations. is this all that should be required?

**Comment by bogdannistorbogdan** [comment]
Hey mate, I'm kinda in the same situation. Did you figure this out?

**Comment by bogdannistorbogdan** [comment]
Hello! I would like to offer more detail regarding my previous comment. I followed the instructions from here:
https://docs.medusajs.com/development/entities/extend-entity
and here:
https://docs.medusajs.com/development/entities/migrations/create

I can see the newly added property in the database. For example, I extended the "Product" entity and added a new field, "brand". Now, how can I have this "Brand" show up in the Admin interface and, obviously, in the store frontend? I am obviously missing the link between what I did and the flow of having the backend and frontend actually show this field.
Also what if I want to completely ignore some fields, like for example, "material". It's in the product table. I really don't need this field for my use case.
Thanks.

**Comment by KareemAbbas1** [comment]
Hello! I'm facing the same issue here. For clearance, after adding allowedStoreProductsRelations and
defaultStoreProductsRelations, I can return the custom field from the store api but not from the admin api. For example, when I send a request to "{baseURL}/store/products/{id}" I get back the newly added custom field, but when I send a request to "{baseURL}/admin/products/{id}" I get back the newly added product without the custom field. Still trying to figure it out, I'd really appreciate it if someone could help us here.

**Reply by KareemAbbas1** [reply]
Just solved it. As mentioned the docs here https://docs.medusajs.com/development/entities/extend-entity#access-custom-attributes-and-relations-in-core-endpoints. You need to add a file under "src/loaders" with the following content:

`export default async function () {
  const imports = (await import(
    "@medusajs/medusa/dist/api/routes/store/products/index"
  )) as any
  imports.allowedStoreProductsFields = [
    ...imports.allowedStoreProductsFields,
    "customAttribute",
  ]
  imports.defaultStoreProductsFields = [
    ...imports.defaultStoreProductsFields,
    "customAttribute",
  ]
}`

This file makes it possible to return the custom field only from the store api. I made a workaround by creating two files under "src/loaders" one named "extend-store-product-field.ts" with the content mentioned in the docs as above, and the other file named "extend-admin-product-field.ts" with the following content:

`export default async function () {
    const imports = (await import(
        "@medusajs/medusa/dist/api/routes/admin/products/index"
    )) as any
    imports.defaultAdminProductFields = [
        ...imports.defaultAdminProductFields,
        "customAttibute",
    ]
}`
which eventually made possible for me to return the custom field from the admin api. Make sure to change "customAttribute" with your attribute.
